### ISSUE TITLE
[security] Version display on all the pages

### ISSUE THREAD

**Issue by mintyPT** [fabiocaccamo/django-admin-interface#153]
**This is not really a bug but it's related to security.**

We recently had a security audit and something that came up was that the name/version/url of this lib was being injected into all the pages of the admin were it's being used (including the login page). The issue is that with this information, an attacker could more easily look for vulnerabilities.

With that said, my ask is quite simple, can we remove this info?

https://github.com/fabiocaccamo/django-admin-interface/blob/master/admin_interface/templates/admin/base_site.html#L15-L16

**Comment by fabiocaccamo** [comment]
@mintyPT I don't catch your point, you could have a project running on python 2.7 + django 1.8 or python 3.10 + django 4.0 both using the latest version of this lib, I don't see the connection between it and a check for vulnerabilities.

*(the injected version is the library version and not python/django version, which could be dangerous to expose)*

**Comment by mintyPT** [comment]
My point is: A good practice is to never disclose the version of anything: nor python, django, nginx or any lib. We never know how something might be exploited and as such, a disclosed version makes it easier for an attacker to find the source code online and look for vulnerabilities. Even worse, if it's something used massively (like django), it's as simple as going to a vulnerability database.

**Comment by fabiocaccamo** [comment]
@mintyPT you convinced me, what do you think about keeping it only if the user is logged-in? In this way I have to change only the template tag.

**Comment by merwok** [comment]
On the other hand, version can be easy to guess (compare checksums of some js file with records of same file in various versions), so hiding version number could provide a false sense of security and not push people to apply more effective techniques (such as updating).

**Comment by mintyPT** [comment]
@fabiocaccamo If the user is logged in it's ok. After all, I guess if an attacker accessed the admin, it doesn't matter anymore.  So, I would be awesome to remove from the template to prevent that bit of info to be publicly visible.

@merwok I fully agree with you! Regularly updating is the thing to do, but in practice it is not always the case. So in this particular situation, the question is: do we want to give free info or not? Even if you update regularly, you shouldn't disclose all the technologies you use.

**Comment by mintyPT** [comment]
Are there other places where this info is visible publicly?

**Comment by fabiocaccamo** [comment]
Probably in some third-party js file.

**Comment by mintyPT** [comment]
I did some looking into the code. The sole situation I found was this kind: `{% static 'admin/css/responsive.css' %}?v={{ version }}`

What is the purpose of adding the version here? Is it meant for caching purposes? 

- - -

Honestly, seeing the version here without context, doesn't shock me too much but, I would still remove/change it (nice to have). For this reason, I looked into the code and an easy way to do that, but still keep the version appended to the link would be to change this

```
@simple_tag(takes_context=False)
def get_admin_interface_version():
    return __version__
```

into something like 

```
import hashlib

@simple_tag(takes_context=False)
def get_admin_interface_version():
    result = hashlib.md5(str(__version__))
    return result.digest()
```

- - -

I can create a PR with the changes discussed.

Let me know if I can go ahead with this.

**Comment by merwok** [comment]
The version tag is certainly there for cache busting when using default static file storages and basic CDN configuration.
It would not be necessary with more advanced configs; could be made controllable with a setting?

**Comment by mintyPT** [comment]
@merwok I looked at ManifestStaticFilesStorage. I guess the responsibility should be left to the base django instead of in a lib or is that a good practice?

@fabiocaccamo Some of the stuff you mentioned has/will be taken care of that is why I started to look into this in the first place.  

So in light of both of those comments, I will go ahead, remove the info discussed and open a PR. If it is accepted awesome, if not, I might as well use the fork instead of the main lib.
